boards/shared/Src/Io/Io_SharedFreqOnlyPwmInput.c

```diff
@@ -0,0 +1,155 @@
+#include <math.h>
+#include <assert.h>
+#include <stm32f3xx_hal.h>
+#include <stdbool.h>
+#include "Io_SharedFreqOnlyPwmInput.h"
+
+struct FreqOnlyPwmInput
+{
+    float frequency_hz;
+
+    TIM_HandleTypeDef *   htim;
+    float                 timer_frequency_hz;
+    uint32_t              rising_edge_tim_channel;
```

Comment:
Is there still any point for the `rising edge` prefix?
```suggestion
    uint32_t              tim_channel;
```

We don't call other things `rising_edge_timer_active_channel`. It feels redundant and a bit confusing to me to have `rising_edge`